### PR TITLE
feat(postgres)!: Update to SDK V3 Arrow native

### PR DESCRIPTION
**Pull request by hermanschaaf**
Closes https://github.com/cloudquery/cloudquery/issues/10763

~~Keeping in draft until we sort out some issues related to internal CQ columns being handled differently in SDK v3~~

BEGIN_COMMIT_OVERRIDE
feat: Update to use [Apache Arrow](https://arrow.apache.org/) type system (#11212)

BREAKING-CHANGE: This release introduces an internal change to our type system to use [Apache Arrow](https://arrow.apache.org/). This should not have any visible breaking changes, however due to the size of the change we are introducing it under a major version bump to communicate that it might have some bugs that we weren't able to catch during our internal tests. If you encounter an issue during the upgrade, please submit a [bug report](https://github.com/cloudquery/cloudquery/issues/new/choose). You will also need to update destinations depending on which one you use:
- Azure Blob Storage >= v3.2.0
- BigQuery >= v3.0.0
- ClickHouse >= v3.1.1
- DuckDB >= v1.1.6
- Elasticsearch >= v2.0.0
- File >= v3.2.0
- Firehose >= v2.0.2
- GCS >= v3.2.0
- Gremlin >= v2.1.10
- Kafka >= v3.0.1
- Meilisearch >= v2.0.1
- Microsoft SQL Server >= v4.2.0
- MongoDB >= v2.0.1
- MySQL >= v2.0.2
- Neo4j >= v3.0.0
- PostgreSQL >= v4.2.0
- S3 >= v4.4.0
- Snowflake >= v2.1.1
- SQLite >= v2.2.0

END_COMMIT_OVERRIDE

### CODE REVIEW

File: plugins/source/postgresql/client/list_tables.go [hunk 1]
```diff
@@ -76,13 +84,22 @@ func (c *Client) listTables(ctx context.Context) (schema.Tables, error) {
...
+			NotNull:    notNull,
+			Unique:     isUnique,
+			Type:       c.PgToSchemaType(columnType),
 		})
 	}
 	return tables, nil
 }
+
+// Add internal columns
+func (c *Client) addInternalColumns(tables []*schema.Table) error {
```
Comment (candiduslynx) [inline plugins/source/postgresql/client/list_tables.go:98]:
```suggestion
func (*Client) addInternalColumns(tables []*schema.Table) error {
```

**Review by candiduslynx — COMMENTED** [review]
some changes indicated by linter

File: plugins/source/postgresql/client/list_tables.go (hunk 1 above)
Comment (erezrokah) [reply plugins/source/postgresql/client/list_tables.go]: We shouldn't need this method in v4 of the SDK so I removed it

**Review by candiduslynx — APPROVED** [review]

File: plugins/source/postgresql/client/list_tables.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/postgresql/client/list_tables.go]: yes, but: https://github.com/cloudquery/cloudquery/actions/runs/5334795623/jobs/9667069012?pr=11212#step:7:68

**Review by erezrokah — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

**Pull request merged**